lib/class/upnp_api.class.php

```diff
@@ -991,13 +1191,317 @@ public static function _videoChilds($prmPath, $prmQuery, $start, $count)
     }
 
     /**
-     * @param $criteria
+     * @param $str
      * @return array
      */
-    public static function _callSearch($criteria)
+    private static function gettokens($str)
     {
-        // Not supported yet
-        return array();
+        /* put the string into lowercase */
+        //    $str = strtolower($str);
+
+        /* make sure ( or ) get picked up as separate tokens */
+        $str = str_replace("(", " ( ", $str);
+        $str = str_replace(")", " ) ", $str);
+
+        /* get the actual tokens */
+        $actualtokens = explode(" ", $str);
+        $actualsize   = sizeof($actualtokens);
+
+        /* trim spaces around tokens and discard those which have only spaces in them */
+        $h=0;
+        for ($i=0; $i < $actualsize; $i++) {
+            $actualtokens[$i]=trim($actualtokens[$i]);
+            if ($actualtokens[$i] != "") {
+                $nospacetokens[$h++] = $actualtokens[$i];
+            }
+        }
+
+        // now put together tokens which are actually one token e.g. upper hutt
+        $onetoken    = "";
+        $h           = 0;
```

Comment:
can we expand this variable name out to something?
i'm not familiar with the hutt word outside star wars?

Reply:
Yes, that's not a very helpful comment.

// now put together the tokens which are not going to be search qualifiers

Would express the intent better